[FEAT] 카드 발급 선택 뷰

AnChanUng opened the pull request:
# 📄 Work Description
- 카드 발급 선택 뷰

# ⚙️ ISSUE
- closed #54 

# 📷 Screenshot
 - 동영상, 사진, 로그 등등
 - ex) 큐알 성공 이미지, 스웨거, 포스트맨 등
 
카드 발급 선택 뷰
![image](https://github.com/user-attachments/assets/74631907-6bbb-4539-b58d-7dc84a73cf43)

 
카드 발급 안내 모달 (CardIssueModal) 
![image](https://github.com/user-attachments/assets/0cb4ad4d-8182-41ec-9b35-7e67d715b87e)


# 💬 To Reviewers
토요일에 못나와서 미안함미다...ㅎㅎㅎㅎㅎㅎ 셤 잘 보고 올게요!!



*** 아직 다 개발 못한 것 **** 
- 카드 캐릭터 얼굴 모양 처럼 나오게 하기 (일요일 저녁에 하겠음) -완료

![image](https://github.com/user-attachments/assets/858976e8-cc10-4a8c-9ec3-ba5dbbdd3cfd)

AnChanUng committed:
[feat/#54]: 전체 틀, 카드 디자인 틀 구현

AnChanUng committed:
[feat/#54]: 카드 캐릭터 컴포넌트 구현


wekids로고, cardChip, cardName(찬웅핑)이 들어있는 카드 컴포넌트를
만들었습니다.

AnChanUng committed:
[file/#54]: CardChip 사진 추가

AnChanUng committed:
[feat/#54]: 색깔 클릭시 카드 배경색 변함

AnChanUng committed:
[feat/#54]: 버튼 색깔 tailwind 구현

AnChanUng committed:
[feat/#54]: 티니핑 캐릭터 누르면 카드 캐릭터 변함

AnChanUng committed:
[feat/#54]: 캐릭터 tailwind 구현

AnChanUng committed:
[feat/#54]: 확인 버튼 구현

AnChanUng committed:
[feat/#54]: 카드 발급 알림 Modal 구현

AnChanUng committed:
Merge branch 'dev' of https://github.com/woorifisa-projects-3rd/WeKids-FE into feat/#54-card_selection

cbj0010 commented on src/app/(card-design)/card-select/page.jsx:
```diff
@@ -0,0 +1,77 @@
+import CardIssueModal from '@/src/ui/Components/card-select/CardIssueModal';
+
+const CardDesignSelector = () => {
+  const [selectedCharacter, setSelectedCharacter] = useState('HEARTSPRING');
```
우리 기본값 파란색 친구로 박는게 좋을 듯!
파란색 친구는 바로핑이고 영어이름으로는 아마 다다핑입니다. 요거 잘 모르겠으면 그 commom에 characterInfoMap이거 보시면 아실 수 있슬겁니다!

cbj0010 commented on src/app/(card-design)/card-select/page.jsx:
```diff
@@ -0,0 +1,77 @@
+        <div className="w-full flex mt-10">
+          <CustomButton size="mediumLarge" className="w-full" onClick={() => setIsModalOpen(true)}>
+            확인
+          </CustomButton>
```
이거 피그마 보시면 large 입니다..

cbj0010 commented on src/ui/Components/card-select/CardCharacter.jsx:
```diff
@@ -0,0 +1,33 @@
+import { characterInfoMap } from '@/src/constants/common';
+
+const CardCharacter = ({ selectedCharacter, selectedColor }) => {
+  const cardName = "찬웅핑";
```
이거 이름도 "찬웅핑"이 고정이 아니라서 얘도 변수로 받아야 될거 같아여!
저거 api 명세 보고 어떤식으로 정보 주는지 확인한 다음에 그 정보 불러왔을 때 꽂아줄 데이터 생각하시면서 변수로 빼고 하는거 생각하시면서 개발 하셔야 되어용!!! 
받아올 더미데이터 하나 만들어 두고 그 변수값을 꽂는걸로 다시 수정해 주세욥

cbj0010 commented on src/app/(card-design)/card-select/page.jsx:
```diff
@@ -0,0 +1,77 @@
+  return (
+    <div className="flex justify-center mt-10">
+      <div className="flex flex-col items-center">
+        <h2 className="text-lg mb-4 text-left w-full">카드 디자인 선택</h2>
```
폰트 사이즈••• 커스텀 한걸로 맞춰주세요..

cbj0010 commented on src/app/(card-design)/card-select/page.jsx:
```diff
@@ -0,0 +1,77 @@
+    <div className="flex justify-center mt-10">
+      <div className="flex flex-col items-center">
+        <h2 className="text-lg mb-4 text-left w-full">카드 디자인 선택</h2>
+        <div className="w-[331px] h-[855.21px] flex-shrink-0 rounded-[10px] border border-black bg-white p-4">
```
소숫점 안들어가게 해주세요..흑흑.. 소숫점 들어간건 반올림합시다

AnChanUng replied:
![image](https://github.com/user-attachments/assets/2389e169-0920-432a-b546-e7cbd2eed57b)
이게 고민이어었던게, h가 935.21로 되어있는데, 그렇게 되면 너무 길어서 임의대로 수정했었습니다.

AnChanUng replied:
피그마에 있는거는 935.21인데, 935로 수정하면 될까요?

cbj0010 replied:
넵! 고렇게 해주세용!

AnChanUng replied:
그렇게 하면 확인 버튼이 잘려용
![image](https://github.com/user-attachments/assets/b41becd8-69e5-48e6-b54c-b479657662c4)

cbj0010 commented on src/app/(card-design)/card-select/page.jsx:
```diff
@@ -0,0 +1,77 @@
+          </div>
+
+          <div className="text-center mb-4">
+            <h3 className="text-md mb-2">카드 캐릭터</h3>
```
폰트사이즈

cbj0010 commented on src/ui/Components/card-select/CardIssueModal.jsx:
pr 올리실때 이런거도 작업한 뷰 캡쳐해서 올려주시면 감사하겠습니다!

cbj0010 commented on src/ui/Components/card-select/ColorButton.jsx:
```diff
@@ -0,0 +1,11 @@
+  return (
+    <button onClick={onClick} className={`w-12 h-12 rounded-full ${colorClass} border border-gray-300`}>
+      
+    </button>
```
색상 테두리색 들어있어요 확인해주세요!

cbj0010 commented on src/ui/Components/card-select/CardCharacter.jsx:
```diff
@@ -0,0 +1,33 @@
+        alt="Card Chip"
+        width={20} 
+        height={20}
+        className="absolute top-10 left-24"
```
칩도 오른쪽으로 더 가야될거 같아용

cbj0010 reviewed (commented):
pr 확인해주세요! 면접 잘 보고 오세용!

AnChanUng committed:
merge dev
Merge branch 'dev' of https://github.com/woorifisa-projects-3rd/WeKids-FE into feat/#54-card_selection

AnChanUng committed:
[refact/#54]: 배경 기본값 다다핑으로 지정

AnChanUng committed:
[refact/#54]: 확인 버튼 크기 Large로 변경

AnChanUng committed:
[refact/#54]: 카드 디자인 선택 text 폰트, css 수정

AnChanUng committed:
[refact/#54]: 전체 틀 w-331, h-935로 수정

AnChanUng committed:
[refact/#54]: 카드 캐릭터 폰트 수정

AnChanUng committed:
[refact/#54]: 배경색 폰트 수정

AnChanUng committed:
[refact/#54]: 티니핑 캐릭터 크기 w-192 h-191로 수정

AnChanUng committed:
[refact/#54]: Card Chip 크기 삭제


Card Chip을 w-31, h-24 (피그마와 같게) 하면 너무 커져서 크기 자체를
지웠습니다. 기존 피그마 사진을 저장한거라 width, height가 없어도 될거라
판단했습니다.

AnChanUng committed:
[refact/#54]: 카드 칩 위치 수정

AnChanUng committed:
[refact/#54]: ColorButton 테두리 수정

AnChanUng committed:
[feat/#54]: 카드이름 "00핑"  zustand, 더미데이터

AnChanUng committed:
[comment#/54]: 불필요한 주석 변경

jabin1221 commented on src/ui/Components/card-select/CardCharacter.jsx:
```diff
@@ -0,0 +1,48 @@
+      <img
+        src="/images/CardChip.svg" 
+        alt="Card Chip"
+        className="absolute top-8 right-1/4 transform -translate-x-1/2 flex-shrink-0"
```
absolute로 되어있던데 혹시 카드의 사이즈가 변하면 칩의 위치도 어색하게 보이나요?

AnChanUng replied:
이게 left-14 left-15 이런식으로 한칸씩 이동하고 싶었는데, 한칸씩 이동 안하고, 규칙 없이 이동해서 다양한 위치 조정중 카드칩 위치가 저기에 갈 수 있는 방법인 absolute right-1/4을 사용했습니다.. 다른 방법으로 저기에 위치시키는 법을 모르겠어요. 불규칙하게 이동해요

jabin1221 reviewed (approved):
고생하셨습니다

AnChanUng committed:
[refact/#54]: 카드 캐릭터 버튼 캐릭터 확대

AnChanUng committed:
[refact/#54]: 카드에 그림자 추가

AnChanUng committed:
[refact/#54]: 캐릭터 버튼 border 수정

AnChanUng committed:
[refact/#54]: 확인 버튼 크기 수정

cbj0010 commented on src/stores/cardStore.js:
```diff
@@ -0,0 +1,15 @@
+import { create } from "zustand";
+
+export const useCardStore = create((set) => ({
+  cardName: '',
+  setCardName: (name) => set({ cardName: name }),
+  fetchCardName: async (id) => {
+    try {
+      const response = await fetch(`/api/cards/${id}`); // 백엔드 API 호출
+      const data = await response.json();
+      set({ cardName: data.cardName });
+    } catch (error) {
+      console.error("Failed to fetch card name:", error);
+    }
+  },
+}));
```
이거는 api 연결부분이여서 service 부분에 들어가는게 맞는거 같습니다!

cbj0010 commented on src/ui/Components/card-select/CharacterButton.jsx:
```diff
@@ -0,0 +1,18 @@
+
+const CharacterButton = ({ character, imagePath, onClick }) => {
+  return (
+    <button 
+      onClick={onClick} 
+      className="w-20 h-20 cursor-pointer overflow-hidden rounded-full border border-black/40"
+    > 
+      <Profile 
+        imagePath={imagePath} 
+        className="w-full h-full object-cover transform scale-150 translate-x-[+00%] translate-y-[+20%]" 
+      />
+    </button>
```
굿
